plugins/qtosgrave/qtosgrave.cpp

```diff
@@ -101,6 +101,15 @@ InterfaceBasePtr CreateInterfaceValidated(InterfaceType type, const std::string&
                     return qtosgrave::CreateQtOSGViewer(penv, sinput); // no idea...
                 }
                 else {
+                    // The function below can be posted before qt app's exec()
+                    // is called, so the function below won't get executed
+                    // immediately and the GetViewer function will time out.
+                    //
+                    // QThread::eventDispatcher(), which can be used to tell if
+                    // an event loop is running in a thread, is only available
+                    // in QT5.
+                    return qtosgrave::CreateQtOSGViewer(penv, sinput);
```

Comment:
Posted functions (in this case QtOSGViewerCreator) will not be executed before the app is started. Therefore, RaveCreateViewer will timeout and fail if it is called before the app is launched.